pkg/types/service_instance.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// Package types contains the Service Manager web entities
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/util"
+)
+
+//go:generate smgen api ServiceInstance
+// ServiceInstance struct
+type ServiceInstance struct {
+	Base
+	Name            string          `json:"name"`
+	ServicePlanID   string          `json:"service_plan_id"`
+	PlatformID      string          `json:"platform_id"`
+	MaintenanceInfo json.RawMessage `json:"maintenance_info"`
+	Context         json.RawMessage `json:"-"`
+	PreviousValues  json.RawMessage `json:"-"`
+	Ready           bool            `json:"ready"`
+	Usable          bool            `json:"usable"`
```

Comment:
What is the semantic of this field

Reply:
When a broker returns an error that the instance is no longer usable (due to an error during async update or deprovision) we also store that information in SM.